src/PsrProducer.php

```diff
@@ -13,4 +13,55 @@
      * @throws InvalidMessageException     if an invalid message is specified
      */
     public function send(PsrDestination $destination, PsrMessage $message);
+
+    /**
+     * Sets the minimum length of time in milliseconds that must elapse after a message is sent before the provider may deliver the message to a consumer.
+     * deliveryDelay is set to zero by default.
+     *
+     * @param float|int $deliveryDelay - the delivery delay in milliseconds.
+     */
+    public function setDeliveryDelay($deliveryDelay);
```

Comment:
While it's a part of JMS, it may not be implementable for some brokers, e.g. RabbitMQ (in default setup) or may require specific broker settings (RabbitMQ will need https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/ plugin to support message dealayed delivery)